codebase2/codebase-sqlite/U/Codebase/Sqlite/Operations.hs

```diff
@@ -1030,9 +1031,28 @@ rebuildNameIndex termNames typeNames = do
   Q.insertTermNames ((fmap (c2sTextReferent *** fmap c2sConstructorType) <$> termNames))
   Q.insertTypeNames ((fmap c2sTextReference <$> typeNames))
 
+data NamesByPath = NamesByPath
+  { termNamesInPath :: [S.NamedRef (C.Referent, Maybe C.ConstructorType)],
+    termNamesExternalToPath :: [S.NamedRef (C.Referent, Maybe C.ConstructorType)],
+    typeNamesInPath :: [S.NamedRef C.Reference],
+    typeNamesExternalToPath :: [S.NamedRef C.Reference]
+  }
+
 -- | Get all the term and type names for the root namespace from the lookup table.
-rootBranchNames :: Transaction ([S.NamedRef (C.Referent, Maybe C.ConstructorType)], [S.NamedRef C.Reference])
-rootBranchNames = do
-  termNames <- Q.rootTermNames
-  typeNames <- Q.rootTypeNames
-  pure (fmap (bimap s2cTextReferent (fmap s2cConstructorType)) <$> termNames, fmap s2cTextReference <$> typeNames)
+rootNamesByPath ::
```

Comment:
Now that I have the namespace in sqlite it's easy to partition names in sqlite which is likely faster than doing it in Haskell.